services/rfq/api/db/api_db.go

```diff
@@ -34,6 +38,187 @@ type Quote struct {
+func (q *ActiveQuoteRequestStatus) Scan(src any) error {
+	res, err := dbcommon.EnumScan(src)
+	if err != nil {
+		return fmt.Errorf("could not scan %w", err)
+	}
+	newStatus := ActiveQuoteRequestStatus(res)
+	*q = newStatus
+	return nil
+}
+
+// Value implements the gorm common interface for enums.
+func (q ActiveQuoteRequestStatus) Value() (driver.Value, error) {
+	// nolint: wrapcheck
+	return dbcommon.EnumValue(q)
+}
+
+var _ dbcommon.Enum = (*ActiveQuoteRequestStatus)(nil)
+
+// ActiveQuoteResponseStatus is the status of a quote request in the db.
+// This is the primary mechanism for moving data through the app.
+//
+// TODO: consider making this an interface and exporting that.
+//
+// EXTREMELY IMPORTANT: DO NOT ADD NEW VALUES TO THIS ENUM UNLESS THEY ARE AT THE END.
+//
+//go:generate go run golang.org/x/tools/cmd/stringer -type=ActiveQuoteResponseStatus
+type ActiveQuoteResponseStatus uint8
+
+const (
+	// Considered means the quote request was considered by the relayer, but was not ultimately the fulfilling response.
+	Considered ActiveQuoteResponseStatus = iota + 1
+	// Returned means the quote request was returned by the relayer to the user.
+	Returned
+	// PastExpiration means the quote request was received, but past the expiration window.
+	PastExpiration
+	// Malformed means that the quote request was malformed.
+	Malformed
+	// Duplicate means that the quote request was a duplicate.
+	Duplicate
+)
+
+// Int returns the int value of the quote request status.
+func (q ActiveQuoteResponseStatus) Int() uint8 {
+	return uint8(q)
+}
+
+// GormDataType implements the gorm common interface for enums.
+func (q ActiveQuoteResponseStatus) GormDataType() string {
+	return dbcommon.EnumDataType
+}
+
+// Scan implements the gorm common interface for enums.
+func (q *ActiveQuoteResponseStatus) Scan(src any) error {
+	res, err := dbcommon.EnumScan(src)
+	if err != nil {
+		return fmt.Errorf("could not scan %w", err)
+	}
+	newStatus := ActiveQuoteResponseStatus(res)
+	*q = newStatus
+	return nil
+}
+
+// Value implements the gorm common interface for enums.
+func (q ActiveQuoteResponseStatus) Value() (driver.Value, error) {
+	// nolint: wrapcheck
+	return dbcommon.EnumValue(q)
+}
+
+var _ dbcommon.Enum = (*ActiveQuoteResponseStatus)(nil)
+
+// ActiveQuoteRequest is the database model for an active quote request.
+type ActiveQuoteRequest struct {
+	RequestID        string                   `gorm:"column:request_id;primaryKey"`
+	UserAddress      string                   `gorm:"column:user_address"`
+	OriginChainID    uint64                   `gorm:"column:origin_chain_id"`
+	OriginTokenAddr  string                   `gorm:"column:origin_token"`
+	DestChainID      uint64                   `gorm:"column:dest_chain_id"`
+	DestTokenAddr    string                   `gorm:"column:dest_token"`
+	OriginAmount     decimal.Decimal          `gorm:"column:origin_amount"`
+	ExpirationWindow time.Duration            `gorm:"column:expiration_window"`
```

Comment:
My thinking here was that it's more precise to specify a window in milliseconds as opposed to a timestamp due to potential clock sync / timezone issues, especially since we may be dealing with very small windows here